cmds/core/dirname/dirname.go

```diff
@@ -9,17 +9,27 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"path/filepath"
 )
 
-func main() {
-	if len(os.Args) < 2 {
-		log.Fatalf("dirname: missing operand")
+func runDirname(w io.Writer, args []string) error {
+	if len(args) < 1 {
+		fmt.Fprintf(w, "%s\n", "dirname: missing operand")
+		return nil
+	}
+	for _, n := range args {
+		if _, err := fmt.Fprintf(w, "%s\n", filepath.Dir(n)); err != nil {
+			return err
+		}
 	}
+	return nil
+}
 
-	for _, n := range os.Args[1:] {
-		fmt.Println(filepath.Dir(n))
+func main() {
+	if err := runDirname(os.Stdout, os.Args[1:]); err != nil {
```

Comment:
interestingly you do not check the len of os.Args here, but you're gonna have to, good news: in the busybox it's not unheard of len(os.Args) to be zero! Further, since os.Args is writeable, an errant init might some day rewrite it, boom! weird failure. I would add code here to clean up os.Args: 
if len(os.Args) == 0 log.Fatal()
runDirname(os.Stdout, os.Args[0], os.Args[1:])

note that with half open ranges, if len(os.Args) is 1, then os.Args[0], os.Args[1:] will work.

https://go.dev/play/p/pYfclHI0C5R